checkov/terraform/checks/resource/aws/S3AllowsAnyPrincipal.py

```diff
@@ -14,29 +15,29 @@ def __init__(self):
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
     def scan_resource_conf(self, conf):
-        if 'policy' in conf.keys():
-            if isinstance(conf['policy'][0], str):
-                try:
-                    policy_block = json.loads(conf['policy'][0])
-                    if 'Statement' in policy_block.keys():
-                        for statement in force_list(policy_block['Statement']):
-                            if statement['Effect'] == 'Deny':
-                                continue
-                            if 'Principal' not in statement:
-                                continue
-
-                            principal = statement['Principal']
-
-                            if principal == '*':
-                                return CheckResult.FAILED
-                            elif 'AWS' in statement['Principal']:
-                                # Can be a string or an array of strings
-                                aws = statement['Principal']['AWS']
-                                if (type(aws) == str and aws == '*') or (type(aws) == list and '*' in aws):
-                                    return CheckResult.FAILED
-                except: # nosec
-                    pass
+        if 'policy' not in conf.keys() or not isinstance(conf['policy'][0], str):
```

Comment:
How does it handle the case where `conf['policy']` is `None` or an empty list?

Reply:
is it possible? I just contained the first two if conditions